### PR TITLE
Routes to get events from conversations and agent messages

### PR DESCRIPTION
**Pull request by philipperolet**

### CODE REVIEW

File: front/pages/api/w/[wId]/assistant/conversations/[cid]/events.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid query parameters, `cId` (string) is required.",
+      },
+    });
+  }
+  const conversationId = req.query.cId;
+  // just checking if the conversation exists, no need to actually construct the
+  // conversation object
+  const conversation = await Conversation.findOne({
```
Comment (spolu) [inline front/pages/api/w/[wId]/assistant/conversations/[cid]/events.ts:64]: Nope no model in API routes

Should be an Result<> pattern on getConversationEvents

Comment (spolu) [reply front/pages/api/w/[wId]/assistant/conversations/[cid]/events.ts:64]: Or a new API in lib/api to do that

Comment (spolu) [reply front/pages/api/w/[wId]/assistant/conversations/[cid]/events.ts:64]: As an example you fail to check the workspace ownership

Let's just get the full conversation that's just simplre

File: front/pages/api/w/[wId]/assistant/conversations/[cid]/messages/[mId]/events.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid query parameters, `cId` (string) is required.",
+      },
+    });
+  }
+  const conversationId = req.query.cId;
+  // just checking if the conversation exists, no need to actually construct the
+  // conversation object
+  const conversation = await Conversation.findOne({
```
Comment (spolu) [inline front/pages/api/w/[wId]/assistant/conversations/[cid]/messages/[mId]/events.ts:67]: Nope same `getConversation`

File: front/pages/api/w/[wId]/assistant/conversations/[cid]/messages/[mId]/events.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid query parameters, `cId` (string) is required.",
+      },
+    });
+  }
+  const messageId = req.query.cId;
+
+  const message = await Message.findOne({
```
Comment (spolu) [inline front/pages/api/w/[wId]/assistant/conversations/[cid]/messages/[mId]/events.ts:94]: Nope no Model

Comment (spolu) [reply front/pages/api/w/[wId]/assistant/conversations/[cid]/messages/[mId]/events.ts:94]: You can find it in the fully constructed conversation from getConversation

**Review by spolu — CHANGES_REQUESTED** [review]

**Comment by philipperolet** [comment]
FYI ~ 20 files in our api routes using .findOne => this is why I was misled

**Comment by philipperolet** [comment]
Fixed and agree it's better this way
however, we have Model.findOne in ~ 20 of our api routes in front so if we believe it's important to avoid, we should refactor that in the future

**Review by spolu — APPROVED** [review]

**Pull request merged**